src/snowflake/snowpark/relational_grouped_dataframe.py

```diff
@@ -253,6 +254,114 @@ def is_valid_tuple_for_agg(e: Union[list, tuple]) -> bool:
+                ---------------------------------------------
+                |"LOCATION"  |"TEMP_C"  |"TEMP_F"           |
+                ---------------------------------------------
+                |SF          |17.5      |63.5               |
+                |SF          |21.0      |69.8               |
+                |SF          |24.0      |75.2               |
+                |NY          |30.9      |87.61999999999999  |
+                |NY          |33.6      |92.48              |
+                ---------------------------------------------
+                <BLANKLINE>
+
+            Call ``apply_in_pandas`` using permanent UDTF with replacing original UDTF:
+
+                >>> from snowflake.snowpark.types import IntegerType, DoubleType
+                >>> _ = session.sql("create or replace temp stage mystage").collect()
+                >>> def group_sum(pdf):
+                ...     pdf.columns = ['grade', 'division', 'value']
+                ...     return pd.DataFrame([(pdf.grade.iloc[0], pdf.division.iloc[0], pdf.value.sum(), )])
+                ...
+                >>> df = session.createDataFrame([('A', 2, 11.0), ('A', 2, 13.9), ('B', 5, 5.0), ('B', 2, 12.1)],
+                ...                              schema=["grade", "division", "value"])
+                >>> df.group_by([df.grade, df.division] ).applyInPandas(
+                ...     group_sum, output_schema=StructType([StructField("grade", StringType()),
+                ...                                        StructField("division", IntegerType()),
+                ...                                        StructField("sum", DoubleType())]),
+                ...                is_permanent=True, stage_location="@mystage", name="group_sum_in_pandas", replace=True
+                ...            ).order_by("sum").show()
+                --------------------------------
+                |"GRADE"  |"DIVISION"  |"SUM"  |
+                --------------------------------
+                |B        |5           |5.0    |
+                |B        |2           |12.1   |
+                |A        |2           |24.9   |
+                --------------------------------
+                <BLANKLINE>
+
+        See Also:
+            - :class:`~snowflake.snowpark.udtf.UDTFRegistration`
+            - :func:`~snowflake.snowpark.functions.pandas_udf`
+        """
+        import pandas as pd
+
+        class _ApplyInPandas:
+            def end_partition(self, pdf: pd.DataFrame) -> pd.DataFrame:
+                return func(pdf)
+
+        # for vectorized UDTF
+        _ApplyInPandas.end_partition._sf_vectorized_input = pd.DataFrame
+
+        # The assumption here is that we send all columns of the dataframe in the apply_in_pandas
+        # function so the inferred input types are the types of each column in the dataframe.
+        inferred_input_types = [field.datatype for field in self._df.schema.fields]
```

Comment:
NIT: `kwargs.get("input_types", [field.datatype for field in self._df.schema.fields])`?

Reply:
added